pkg/kubelet/status/status_manager.go

```diff
@@ -566,43 +585,44 @@ func (m *manager) syncBatch() {
 			}
 		}
 
+		// Calculate all possible status updates
 		for uid, status := range m.podStatuses {
-			syncedUID := kubetypes.MirrorPodUID(uid)
-			if mirrorUID, ok := podToMirror[kubetypes.ResolvedPodUID(uid)]; ok {
-				if mirrorUID == "" {
-					klog.V(5).InfoS("Static pod does not have a corresponding mirror pod; skipping",
-						"podUID", uid,
-						"pod", klog.KRef(status.podNamespace, status.podName))
-					continue
-				}
-				syncedUID = mirrorUID
-			}
-			if m.needsUpdate(types.UID(syncedUID), status) {
-				updatedStatuses = append(updatedStatuses, podStatusSyncRequest{uid, status})
-			} else if m.needsReconcile(uid, status.status) {
-				// Delete the apiStatusVersions here to force an update on the pod status
-				// In most cases the deleted apiStatusVersions here should be filled
-				// soon after the following syncPod() [If the syncPod() sync an update
-				// successfully].
-				delete(m.apiStatusVersions, syncedUID)
```

Comment:
This construct is no longer necessary in the new code, because syncPod no longer calls needUpdate to check this value and we directly invoke syncPod only if needed.